### PR TITLE
NStack refactor

### PR DESCRIPTION
**Pull request by jcmsalves**
- TranslationOptions moved out of translation manager
- Remove singletons -> access through NStack
- Make resource files private
- save()/load() in app open settings -> cachemanager
- Move all PrefsManager code to CacheManager

### CODE REVIEW

File: app/src/main/java/dk/nodes/nstack/NStack.java [hunk 1]
```diff
@@ -78,23 +91,31 @@ public Context getApplicationContext() {
...
         }
 
         return appOpenManager;
     }
 
+    public TranslationOptions getTranslationOptions() {
+        return translationOptions;
+    }
+
+    public BackendManager getBackendManager() {
```
Comment (johsoe) [inline app/src/main/java/dk/nodes/nstack/NStack.java]: I think we need to kill this and instead pass in the reference to this where it's needed, ie. AppOpenManager which already takes TranslationManager in the constructor.

File: app/src/main/java/dk/nodes/nstack/NStack.java [hunk 2]
```diff
@@ -78,23 +91,31 @@ public Context getApplicationContext() {
     }
 
     public TranslationManager getTranslationManager() {
-        return TranslationManager.getInstance();
+        return translationManager;
     }
 
     public AppOpenManager getAppOpenManager() {
```
Comment (johsoe) [inline app/src/main/java/dk/nodes/nstack/NStack.java]: I think we need to kill this and instead pass in the reference to this where it's needed, ie. AppOpenManager which already takes TranslationManager in the constructor.

File: app/src/main/java/dk/nodes/nstack/NStack.java [hunk 3]
```diff
@@ -78,23 +91,31 @@ public Context getApplicationContext() {
     }
 
     public TranslationManager getTranslationManager() {
```
Comment (johsoe) [inline app/src/main/java/dk/nodes/nstack/NStack.java]: I think we need to kill this and instead pass in the reference to this where it's needed, ie. AppOpenManager which already takes TranslationManager in the constructor.

File: app/src/main/java/dk/nodes/nstack/util/appopen/AppOpenManager.java [hunk 4]
```diff
@@ -46,22 +42,29 @@
...
 
-    public AppOpenManager(  ) {
+    private Context context;
+    private CacheManager cacheManager;
+    private TranslationManager translationManager;
+
+    public AppOpenManager(Context context, TranslationManager translationManager) {
+        this.context = context;
+        this.translationManager = translationManager;
+        cacheManager = new CacheManager(context);
```
Comment (johsoe) [inline app/src/main/java/dk/nodes/nstack/util/appopen/AppOpenManager.java]: Shouldn't we initialize this in NStack constructor or?

File: app/src/main/java/dk/nodes/nstack/util/translation/TranslationManager.java [hunk 5]
```diff
@@ -395,26 +285,26 @@ private void updateTranslationLanguageKeys(JSONObject data) {
...
-                // Selected locale doesnt exist, continue to fallback
-                if (!localeExists && fallbackLocaleExists && !languageName.equalsIgnoreCase(translationOptions.getFallbackLocale())) {
-                    continue;
-                }
-
-                // fallback doesnt exist either, continue until we find something that matches fallbacks, ie: en-**
-                if (!localeExists && !fallbackLocaleExists && !translationOptions.getFallbackLocale().startsWith(languageName.substring(0, 2))) {
-                    continue;
-                }
+//                // Only update current language, if we have more than one language
```
Comment (johsoe) [inline app/src/main/java/dk/nodes/nstack/util/translation/TranslationManager.java]: Remove or leave it, not this 👀

File: app/src/main/java/dk/nodes/nstack/util/translation/TranslationAPIClient.java [hunk 6]
```diff
@@ -0,0 +1,129 @@
...
+import dk.nodes.nstack.util.model.Language;
+import okhttp3.Call;
+import okhttp3.Callback;
+import okhttp3.Response;
+
+/**
+ * Created by joaoalves on 13/12/2016.
+ */
+
+public class TranslationAPIClient {
```
Comment (johsoe) [inline app/src/main/java/dk/nodes/nstack/util/translation/TranslationAPIClient.java]: Still not sold on this name, TranslationClient? Suggestions?

Comment (sanogueralorenzo) [inline app/src/main/java/dk/nodes/nstack/util/translation/TranslationAPIClient.java]: both are ugly names, TranslationBackendManager sounds better

Comment (johsoe) [inline app/src/main/java/dk/nodes/nstack/util/translation/TranslationAPIClient.java]: Best one yet

File: app/src/main/java/dk/nodes/nstack/util/appopen/AppOpenManager.java (hunk 4 above)
Comment (jcmsalves) [inline app/src/main/java/dk/nodes/nstack/util/appopen/AppOpenManager.java]: done

File: app/src/main/java/dk/nodes/nstack/NStack.java (hunk 3 above)
Comment (jcmsalves) [inline app/src/main/java/dk/nodes/nstack/NStack.java]: deleted, was not being used already

File: app/src/main/java/dk/nodes/nstack/NStack.java (hunk 2 above)
Comment (jcmsalves) [inline app/src/main/java/dk/nodes/nstack/NStack.java]: same as above

File: app/src/main/java/dk/nodes/nstack/util/translation/TranslationManager.java (hunk 5 above)
Comment (jcmsalves) [inline app/src/main/java/dk/nodes/nstack/util/translation/TranslationManager.java]: Left it there as it might be needed when we need to get all languages

**Pull request merged**